sentinelhub/geopedia.py

```diff
@@ -90,61 +101,87 @@ def session_id(self):
-        return self.get_session_id(start_new=False)
+        return self._provide_session()['sessionId']
+
+    @property
+    def session_headers(self):
+        """ Headers which have to be used when accessing any data from Geopedia with this session
+
+        :return: A dictionary containing session headers
+        :rtype: dict
+        """
+        return {
+            self._provide_session()['sessionHeaderName']: self.session_id
+        }
+
+    @property
+    def user_info(self):
+        """ Information that this session has about user
+
+        :return: A dictionary with user info
+        :rtype: dict
+        """
+        return self._provide_session()['user']
+
+    @property
+    def user_id(self):
+        """ Geopedia user ID. If no login was done during session this will return `'NO_USER'`.
 
-    def get_session_id(self, start_new=False):
+        :return: User ID string
+        :rtype: str
+        """
+        return self.user_info['id']
+
+    def restart(self):
+        """ Method that restarts Geopedia Session
+
+        :return: It returns the object itself, with new session
+        :rtype: GeopediaSession
+        """
+        self._provide_session(start_new=True)
+        return self
+
+    def _provide_session(self, start_new=False):
         """ Returns a session ID
 
         :param start_new: If `True` it will always create a new session. Otherwise it will create a new
             session only if no session exists or the previous session timed out.
         :type start_new: bool
-        :return: A session ID string
-        :rtype: str
+        :return: Current session info
+        :rtype: dict
         """
         if self.is_global:
-            self._session_id = self._global_session_id
-            self._session_end = self._global_session_end
+            self._session_info = self._global_session_info
+            self._session_start = self._global_session_start
 
-        if self._session_id is None or start_new or datetime.datetime.now() > self._session_end:
+        if self._session_info is None or start_new or \
+                datetime.datetime.now() > self._session_start + self.SESSION_DURATION:
             self._start_new_session()
 
-        return self._session_id
+        return self._session_info
 
     def _start_new_session(self):
-        """ Updates the session id and calculates when the new session will end. If username and password are provided
+        """ Starts a new session and calculates when the new session will end. If username and password are provided
         it will also make login.
         """
-        self._session_end = datetime.datetime.now() + self.SESSION_DURATION
+        self._session_start = datetime.datetime.now()
 
         session_url = '{}data/v1/session/create?locale=en'.format(self.base_url)
-        self._session_id = get_json(session_url)['sessionId']
+        self._session_info = get_json(session_url)
 
         if self.username and self.password:
```

Comment:
I would add a check on whether either are specified, something like

```python
if self.username and self.password:
    self._make_login()
elif self.username or self.password:
    raise ValueError("Both username and password must be specified at login")
```

Reply:
Don't forget that user might have specified password_md5 (and not password)

Reply:
Attributes `password` and `password_md5` are actually put into the same object `self.password` in `__init__`. At the moment if someone would for some reason specify both, `password` would have priority over `password_md5`.

I think current username/password handling is intuitive but just in case I can add in `__init__`:
- If someone specifies `password` and `password_md5` and error is raised.
- If only one of `self.username` and `self.password` are available an error is raised.

Reply:
Added in https://github.com/sentinel-hub/sentinelhub-py/pull/56/commits/b48e7703a157531fedbaf8a6637c8e1a91b5b8d6